packages/gestalt/src/DropdownItem.js

```diff
@@ -0,0 +1,92 @@
+// @flow strict
+import React, { type Node } from 'react';
+import PropTypes from 'prop-types';
+import MenuOption, { type OptionObject } from './MenuOption.js';
+import DropdownContext from './DropdownContextProvider.js';
+
+type PublicProps = {|
+  badgeText?: string,
+  children?: Node,
+  handleSelect?: ({|
+    event: SyntheticFocusEvent<HTMLInputElement>,
+    item: OptionObject,
+  |}) => void,
+  isExternal?: boolean,
+  option: OptionObject,
+  selected?: OptionObject | $ReadOnlyArray<OptionObject> | null,
+  href?: string,
+|};
+
+type PrivateProps = {|
+  index?: number,
+|};
+
+type Props = {|
+  ...PublicProps,
+  ...PrivateProps,
+|};
+
+export default function DropdownItem({
+  badgeText,
+  children,
+  handleSelect,
+  index = 0,
+  isExternal,
+  option,
+  selected,
+  href,
+}: Props): Node {
+  return (
+    <DropdownContext.Consumer>
+      {({ hoveredItem, setHoveredItem, setOptionRef }) => (
+        <MenuOption
+          key={`${option.value + index}`}
```

Comment:
Ah, just noticed this: we should avoid using the `+` for string concatenation, as it can lead to subtle and unintuitive bugs. There's an eslint rule for this; I'll add it to our meeting topics this week.